File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        cur_price = prices[0]
        max_price = 0
        
        for price in prices:
            if price < cur_price:
                cur_price = price
            else:
                max_price = max(max_price, price - cur_price)
        
        return max_price

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int curr = prices[0];
        int total = 0;

        for(int price: prices){
            if(curr < price){
                total = max(total, price-curr);
            }
            else{
                curr = price;
            }
        }

        return total;
    }
};